std/bitmanip.d

```diff
@@ -799,18 +801,157 @@ private:
     }
 
 public:
-    /**********************************************
-     * Gets the amount of native words backing this $(D BitArray).
-     */
-    @property size_t dim() const @nogc pure nothrow @safe
+    /**
+    Creates a `BitArray` from a `bool` array, such that `bool` values read
+    from left to right correspond to subsequent bits in the `BitArray`.
+
+    Params: ba = Source array of `bool` values.
+    */
+    this(in bool[] ba) nothrow pure
+    {
+        length = ba.length;
+        foreach (i, b; ba)
+        {
+            this[i] = b;
+        }
+    }
+
+    ///
+    @system unittest
+    {
+        import std.algorithm.comparison : equal;
+
+        auto a = BitArray([1, 0, 0, 1, 1]);
+
+        assert(a.length == 5);
+        assert(a.bitsSet.equal([0, 3, 4]));
+    }
+
+    /**
+    Creates a `BitArray` from the raw contents of the source array. The
+    source array is not copied but simply acts as the underlying array
+    of bits, which stores data as `size_t` units.
+
+    That means a particular care should be taken when passing an array
+    of a type different than `size_t`, firstly because its length should
+    be a multiple of `size_t.sizeof`, and secondly because how the bits
+    are mapped:
+    ---
+    size_t[] source = [1, 2, 3, 3424234, 724398, 230947, 389492];
+    auto sbits = size_t.sizeof * 8;
+    auto ba = BitArray(source, source.length * sbits);
+    foreach (n; 0 .. source.length * sbits)
+    {
+        auto nth_bit = cast(bool) (source[n / sbits] & (1L << (n % sbits)));
+        assert(ba[n] == nth_bit);
+    }
+    ---
```

Comment:
This example should be added 1:1 as a unittest,s.t. it can't get broken by future released.

Reply:
Yeah and actually it's already there (3rd example for this constructor).